bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,1135 @@
+        if (contentType != null) {
+            fields.put(HttpHeader.CONTENT_TYPE, contentType);
+        }
+        if (contentLength >= 0) {
+            fields.putLongField(HttpHeader.CONTENT_LENGTH, contentLength);
+        }
+        fields.put(APPLICATION_KEY, applicationKey);
+        return new HeadersFrame(new MetaData.Request(method, new HttpURI(url), HttpVersion.HTTP_2, fields), null,
+                contentLength <= 0);
+    }
+
+    /**
+     * Use an HTTP/2 PUT command to send a resource to the server.
+     *
+     * @param resource the resource to put.
+     * @throws ApiException if something fails.
+     * @throws InterruptedException
+     */
+    public void putResource(Resource resource) throws ApiException, InterruptedException {
+        sleepDuringRestart();
+        if (onlineState == State.CLOSED) {
+            return;
+        }
+        Session session = http2Session;
+        if (Objects.isNull(session) || session.isClosed()) {
+            throw new ApiException("HTTP 2 session is null or closed");
+        }
+        throttle();
+        String requestJson = jsonParser.toJson(resource);
+        ByteBuffer requestBytes = ByteBuffer.wrap(requestJson.getBytes(StandardCharsets.UTF_8));
+        String url = getUrl(new ResourceReference().setId(resource.getId()).setType(resource.getType()));
+        HeadersFrame headers = prepareHeaders(url, MediaType.APPLICATION_JSON, "PUT", requestBytes.capacity(),
+                MediaType.APPLICATION_JSON);
+        logger.trace("PUT {} HTTP/2 >> {}", url, requestJson);
+        try {
+            Completable<@Nullable Stream> streamPromise = new Completable<>();
+            ContentStreamListenerAdapter contentStreamListener = new ContentStreamListenerAdapter();
+            session.newStream(headers, streamPromise, contentStreamListener);
+            // wait for stream to be opened
+            Stream stream = Objects.requireNonNull(streamPromise.get(TIMEOUT_SECONDS, TimeUnit.SECONDS));
+            stream.data(new DataFrame(stream.getId(), requestBytes, true), Callback.NOOP);
+            // wait for HTTP response
+            String contentJson = contentStreamListener.awaitResult();
+            String contentType = contentStreamListener.getContentType();
+            logger.trace("HTTP/2 200 OK (Content-Type: {}) << {}", contentType, contentJson);
+            if (!MediaType.APPLICATION_JSON.equals(contentType)) {
+                throw new ApiException("Unexpected Content-Type: " + contentType);
+            }
+            try {
+                Resources resources = Objects.requireNonNull(jsonParser.fromJson(contentJson, Resources.class));
+                if (logger.isDebugEnabled()) {
+                    resources.getErrors().forEach(error -> logger.debug("putResource() resources error:{}", error));
+                }
+            } catch (JsonParseException e) {
+                logger.warn("putResource() error parsing JSON response:{}", contentJson);
```

Comment:
This warning is redundant, since the exception is caught by `handleCommand` and also logged as warning there. You could add `contentJson` to the exception if needed.

Reply:
I will move it to debug